clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -2047,6 +2047,14 @@ StringRef CodeGenModule::getMangledName(GlobalDecl GD) {
                  GD.getWithKernelReferenceKind(KernelReferenceKind::Kernel),
                  ND));
 
+  // This invariant should hold true in the future.
+  // Prior work:
+  // https://discourse.llvm.org/t/rfc-clang-diagnostic-for-demangling-failures/82835/8
+  // https://github.com/llvm/llvm-project/issues/111345
+  // assert(llvm::isMangledName(MangledName) &&
```

Comment:
isMangledName doesn't seem to exist?

But, also, we'd only want to validate output actually produced by Clang `mangleCXXName`, and not e.g. a user-written `void f() asm("_ZGARBAGE"); void f() {}`

Maybe we want:`if ((MangledName.startswith("_Z") || MangledName.startswith("?")) & !GD->hasAttr<AsmLabelAttr>())`?

Reply:
No, it doesn't exist. This was to demonstrate the idea.

Thanks, sure, I can change the text.